libraries/chain/include/eosio/chain/abi_serializer.hpp

```diff
@@ -970,5 +972,73 @@ void abi_serializer::from_variant( const fc::variant& v, T& o, Resolver resolver
    from_variant( v, o, resolver, create_yield_function(max_serialization_time) );
 }
 
+using abi_serializer_cache = std::unordered_map<account_name, std::optional<abi_serializer>>;
+   
+class abi_resolver {
+public:
+   abi_resolver(abi_serializer_cache&& abi_cache) :
+      abi_cache(std::move(abi_cache))
+   {}
+
+   std::optional<std::reference_wrapper<abi_serializer>> operator()(const account_name& account) {
+      auto it = abi_cache.find(account);
+      if (it != abi_cache.end() && it->second)
+         return std::reference_wrapper<abi_serializer>(*it->second);
+      return {};
+   };
+
+private:
+   abi_serializer_cache abi_cache;
```

Comment:
Probably better named to `serializer_cache`?

Reply:
Done. I actually renamed it `abi_serializers`.

Reply:
Hmm, I think I like `abi_cache` better than `abi_serializers`. Maybe `serializer_abi_cache`?